pkg/storage/prunable/retainer.go

```diff
@@ -1,162 +1,183 @@
 package prunable
 
 import (
-	"github.com/iotaledger/hive.go/core/storable"
-	"github.com/iotaledger/hive.go/ds/types"
 	"github.com/iotaledger/hive.go/ierrors"
 	"github.com/iotaledger/hive.go/kvstore"
 	"github.com/iotaledger/hive.go/lo"
+	"github.com/iotaledger/hive.go/serializer/v2/marshalutil"
 	iotago "github.com/iotaledger/iota.go/v4"
 	"github.com/iotaledger/iota.go/v4/nodeclient/apimodels"
 )
 
 const (
-	blockOrphanedPrefix byte = iota
-	blockConfirmedPrefix
-	blockFailurePrefix
-	transactionPendingPrefix
-	transactionConfirmedPrefix
-	transactionFailurePrefix
+	blockStorePrefix byte = iota
+	transactionStorePrefix
 )
 
+type BlockRetainerData struct {
+	State         apimodels.BlockState
+	FailureReason apimodels.BlockFailureReason
+}
+
+func (b *BlockRetainerData) Bytes() ([]byte, error) {
+	marshalUtil := marshalutil.New()
```

Comment:
Can we initialize the util with the size already? Could reduce allocation operations.